packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -232,14 +237,38 @@ export abstract class AbstractMessageManager<
    * @param rawSig - The raw data of the signature request.
    */
   setMessageStatusSigned(messageId: string, rawSig: string) {
+    this.setMessageStatusAndResult(messageId, rawSig, 'signed');
+  }
+
+  /**
+   * Sets a Message status to 'decrypted' via a call to this.setMessageStatus and updates
+   * that Message in this.messages by adding the raw signature data of the signature
+   * request to the Message.
+   *
+   * @param messageId - The id of the Message to sign.
+   * @param rawSig - The raw data of the signature request.
+   */
+  setMessageStatusDecrypted(messageId: string, rawSig: string) {
```

Comment:
I set up the `setMessageStatusAndResult` so that I could call that method directly from the controller instead of adding more methods for each state that we add.

Having said that, there's the argument that creating a method stops a dev from passing an invalid state, although the plan is to make that extra difficult (for TypeScript) by adding types for the states in the future.

If we decide to go with this approach, I'll have to create a method like this in my PR too, but I'm happy either way.

Reply:
Ok, this makes sense, I will hold my PR until yours is merged anyway then will rebase it.